src/include/stir/recon_buildblock/PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData.txx

```diff
@@ -159,9 +181,137 @@ post_processing()
+      if (header.parse(_normalisations_filename.c_str()) == false)
+          error("PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData::post_processing Error parsing %s", _normalisations_filename.c_str());
+      _normalisations_sptr.resize(header.get_num_data_sets());
+      for (int i=0; i<header.get_num_data_sets(); ++i)
+          _normalisations_sptr[i].reset(
+                      new BinNormalisationFromProjData(header.get_filename(i)));
+      // Check there are as many binnormalisations as sinograms
+      if (_normalisations_sptr.size() != _dyn_proj_data_sptr->get_num_proj_data())
+          error("Mismatch in # sinograms and # normalisation sinograms.");
+  }
+
+  // motion
+  set_up_motion();
+
   return false;
 }
 
+template<typename TargetT>
+void
+PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData<TargetT>::
+set_up_motion()
+{
+    // Check how motion was entered
+    // vector of filenames, where 0th is combined (e.g.,4D nifti),
+    // and 1st, 2nd & 3rd are x-, y-, & z-components
+    bool combined_motion_present  = (_disp_field_filenames[0] != "0");
+    bool x_motion_present         = (_disp_field_filenames[1] != "0");
+    bool y_motion_present         = (_disp_field_filenames[2] != "0");
+    bool z_motion_present         = (_disp_field_filenames[3] != "0");
+
+    // If no motion, return
+    if (!combined_motion_present && !x_motion_present && !y_motion_present && !z_motion_present) {
+        info(boost::format("No motion information entered"));
+        return;
+    }
+
+    // If part of the individual motion has been entered
+    if (x_motion_present != y_motion_present != z_motion_present)
+        error(boost::format("Some but not all individual components have been entered. Enter all or none."));
+
+    // If motion has been entered two different ways (4D and 3 separate components)
+    if (combined_motion_present && x_motion_present && y_motion_present && z_motion_present)
+        error(boost::format("Both combined (4D) and separated (x, y and z) motion has been entered. Enter one or the other."));
+
+    // Read the files
+    std::vector<MultipleDataSetHeader> disp_field_headers;
+
+    // If combined
+    if (combined_motion_present) {
+        disp_field_headers.resize(1);
+        if (disp_field_headers[0].parse(_disp_field_filenames[0].c_str()) == false)
+            error("PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData::post_processing Error parsing %s", _disp_field_filenames[0].c_str());
+    }
+    // If separate components
+    else {
+        disp_field_headers.resize(3);
+        for (int i=0; i<3; ++i)
+            if (disp_field_headers[i].parse(_disp_field_filenames[i+1].c_str()) == false)
+                error("PoissonLogLikelihoodWithLinearKineticModelAndDynamicProjectionData::post_processing Error parsing %s", _disp_field_filenames[i+1].c_str());
+    }
+
+    // Check there are as many motion fields as sinograms
```

Comment:
Is this required? 

I suppose that at some point we might not want to have 1 DVF == 1 sino, but may want to reuse some DVFs?

Reply:
yes, but re-using becomes a bit tricky how to enter it which frame to re-use where. That would all sit in the input part, which arguably sits here. Until we would do that, I would leave this test in.